### DISCUSSION TITLE
.NET support for 7z + RAR per Windows 11 Explorer support?

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [dotnet/runtime#92763] [Q&A] [answered]
**@Symbai, https://github.com/dotnet/runtime/issues/1542#issuecomment-1738598132**

> The Windows 11 update which supports RAR and 7zip has been released. How/when can we use that in .NET ?

**Comment by Symbai** [comment, marked as answer]
I've successfully extracted both in Windows 11. What I really think is funny is that they act like a folder. In one of my existing .NET applications which displays shell items in a treelist. I now can expand a RAR file and see its content... its treated as a folder. The .NET app is built with .NET 6.

![image](https://github.com/dotnet/runtime/assets/14368203/eab05d2a-04da-4b4c-80f5-b30b971f5270)

//edit: A ZIP file doesn't behave like that. I dont know if that is a bug or intended behavior.

![image](https://github.com/dotnet/runtime/assets/14368203/c566b4c8-fa1f-4f73-a55d-6c85e240fdd3)
